aider/main.py

```diff
@@ -557,18 +564,86 @@ def scrub_sensitive_info(text):
-    if args.openai_api_type == "azure":
+    if not args.model:
+        args.model = default_model
+
+    if args.litellm:
+        if LITELLM_SPEC is None:
+            io.tool_error("LiteLLM is not installed. Install it with `pip install litellm`.")
+            return 1
+
+        io.tool_output("LiteLLM is enabled.")
+
+        packageRootDir = os.path.dirname(LITELLM_SPEC.origin)
+        modelPricesBackupName = "model_prices_and_context_window_backup.json"
+        modelPricesPath = os.path.join(packageRootDir, modelPricesBackupName)
+
+        # LiteLLM keeps a backup of the model prices for when the network is
+        # down, but it's otherwise not used unless LITELLM_LOCAL_MODEL_COST_MAP
+        # is set to True. To keep Aider's startup time fast, prevent the network
+        # request unless the model prices haven‘t been updated in 12 hours.
+        from datetime import datetime, timedelta
+        timeSinceModified = datetime.now() - datetime.fromtimestamp(os.path.getmtime(modelPricesPath))
+
+        if timeSinceModified < timedelta(hours=12):
+            os.environ["LITELLM_LOCAL_MODEL_COST_MAP"] = "True"
+            io.tool_output("LiteLLM model list backup is recent, using it instead of network request.")
+        else:
+            def strfdelta(delta):
+                days = delta.days
+                hours, remainder = divmod(delta.seconds, 3600)
+                minutes, seconds = divmod(remainder, 60)
+                time_parts = [f"{days} day{'' if days == 1 else 's'}" if days else "",
+                                f"{hours} hour{'' if hours == 1 else 's'}" if hours else "",
+                                f"{minutes} minute{'' if minutes == 1 else 's'}" if minutes else "",
+                                f"{seconds} second{'' if seconds == 1 else 's'}"]
+                time_parts = [part for part in time_parts if part]
+                return ", ".join(time_parts)
+
+            formattedTime = strfdelta(timeSinceModified)
+            io.tool_output(f"LiteLLM model list backup last updated {formattedTime} ago, updating now.")
+
+            # Since LiteLLM never updates its backup after hitting the
+            # network, we'll have to do it ourselves.
+            from litellm import model_cost
+            with open(modelPricesPath, "w") as f:
+                import json
+                json.dump(model_cost, f)
+
+        # Strangely, LiteLLM has DEBUG-level logging enabled by default. This is
+        # a rather fragile solution, so we should report the issue upstream.
+        import logging
+        from litellm._logging import handler
+        handler.setLevel(logging.ERROR)
+
+        # Support LITELLM_API_KEY, LITELLM_BASE_URL, etc.
+        litellm_kwargs = {
+            key.replace("LITELLM_", "").lower(): value
+            for key, value in os.environ.items()
+            if key.startswith("LITELLM_")
+        }
+
+        from litellm import LiteLLM
+        client = LiteLLM(**litellm_kwargs)
```

Comment:
@nkeilar please try setting `LITELLM_BASE_URL` environment variable and let me know if that works for you